src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs

```diff
@@ -628,126 +628,187 @@ protected AnalyzerActions GetOrCreateAnalyzerActions(DiagnosticAnalyzer analyzer
     // moves from an analyzer-centric model to an action-centric model. For example, the driver would need to stop asking
     // if a particular analyzer can analyze syntax trees, and instead ask if any syntax tree actions are present. Also,
     // the driver needs to apply all relevant actions rather then applying the actions of individual analyzers.
-    internal sealed class AnalyzerActions
+    internal struct AnalyzerActions
```

Comment:
Can this be a readonly struct with `IsEmpty` and Concurrent moved to a With-ing pattern?

Reply:
No, this is still a mutable struct. It would be a decent size change to convert all the remaining mutation methods to immutable transforms.